#!/usr/bin/env python3
PACKAGE = "ros_cv_tools"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Define the sliders for HSV range
gen.add("h_low", int_t, 0, "Hue low value", 0, 0, 180)
gen.add("h_high", int_t, 0, "Hue high value", 180, 0, 180)
gen.add("s_low", int_t, 0, "Saturation low value", 0, 0, 255)
gen.add("s_high", int_t, 0, "Saturation high value", 255, 0, 255)
gen.add("v_low", int_t, 0, "Value low value", 0, 0, 255)
gen.add("v_high", int_t, 0, "Value high value", 255, 0, 255)
gen.add("white_percent", bool_t, 0, "Publish percentage of white pixels", True)

exit(gen.generate(PACKAGE, "ros_cv_tools", "HSV"))
